### PR TITLE
Updating inspec with bastion options

### PR DESCRIPTION
**Pull request by frezbo**
Updating inspec with bastion options as per https://github.com/inspec/train/pull/310

Signed-off-by: Noel Georgi <18496730+frezbo@users.noreply.github.com>

@jquick I will update the Gemspec once train is released with a new version. Can this go in the next release of inspec? Thanks

### CODE REVIEW

**Comment by frezbo** [comment]
@jquick I see the functional test failure, which I do not see locally, is the test environment doing an integration test? Should I remove that?

**Comment by jquick** [comment]
Hey @frezbo, the release going out today will automatically pull the latest train in the allowed range. So  it will auto have the train changes already released. 

Yes it does do a functional test. This unfortunately wont make this morning release but will be released next week.

File: test/functional/inspec_exec_test.rb [hunk 1]
```diff
@@ -300,6 +300,14 @@
     end
   end
 
+  describe 'when --bastion-host and --proxy_command is used' do
+    it 'raises an exception when both flags are provided' do
+      out = inspec('exec ' + example_profile + '  -t ssh://dummy@dummy --proxy_command dummy --bastion_host dummy')
```
Comment (jquick) [inline test/functional/inspec_exec_test.rb]: probebly just add a `--password dummy` to the command, to ignore the ssh errors

**Comment by frezbo** [comment]
I think this should be good to go now :+1:

**Comment by frezbo** [comment]
@jquick can this be merged now, I hope this goes into next release, Thanks for the support :+1:

**Review by jquick — APPROVED** [review]
Thanks @frezbo

**Review by jerryaldrichiii — APPROVED** [review]
Looks good to me! Thanks @frezbo!

File: .gitignore [hunk 2]
```diff
@@ -22,6 +22,7 @@ habitat/results
 /.direnv
 /.envrc
 results/
+TAGS
```
Comment (jquick) [inline .gitignore]: Can you remove this? feel free to keep your local copy but we should not be changing the global `.gitignore` unless inspec is generating something into `TAGS` for everyone.

Comment (frezbo) [reply .gitignore]: @jquick  done

**Comment by frezbo** [comment]
@jquick the failure seems to be a  fluke

**Comment by jerryaldrichiii** [comment]
Rerunning Travis to verify, but I think you're right @frezbo

**Comment by frezbo** [comment]
@jerryaldrichiii looks like it failed again, mind taking a look

**Comment by frezbo** [comment]
the tests are failing locally too, did anything change upstream, it was working fine when I tested last time.

**Pull request merged**